mewarpx/mewarpx/runinfo.py

```diff
@@ -0,0 +1,479 @@
+                if shape.name == '':
+                    raise ValueError("No name assigned for shape {}".format(
+                        shape))
+                if check_names:
+                    if shape.name in self._reserved_names:
+                        raise ValueError(("Repeated or reserved name {} used "
+                                          "for shape {}").format(
+                                            shape.name, shape))
+                    self._reserved_names.append(shape.name)
+                if shape.WF == 0.0:
+                    raise ValueError(("Work function for shape {} not assigned"
+                                      "; assign WF to this shape.").format(
+                                        shape.name))
+                if mwxrun.me == 0:
+                    if callable(shape.voltage):
+                        voltage = shape.voltage(mwxrun.get_t())
+                    else:
+                        voltage = shape.voltage
+                    print(("Saved shape {} of type {} with WF {} eV and voltage"
+                           " {:.2f} V (at t = {:.2f} ns)").format(
+                                shape.name, component, shape.WF, voltage,
+                                mwxrun.get_t()*1e9)
+                          )
+
+        V_cathode, WF_cathode = self._get_cathode_voltage_wf()
+
+    def _get_patchy_cathode(self):
+        # Check if a patchy cathode is used by checking if "CATHODE_DELTA_PHI"
+        # is greater than 0.0
+        return (self.user_var_dict.get("CATHODE_DELTA_PHI", 0.) > 0.)
+
+    def _get_cathode_voltage_wf(self):
+        # All cathode shapes must have same V and WF for accurate power
+        # calculation unless the cathode is patchy in which case the
+        # average voltage is used.
+        return get_voltage_wf(self.surface_dict['cathode'],
+                              patchy_cathode=self._get_patchy_cathode())
+
+    def _save_sim_info(self):
+        """Sets up the run_param_dict with helpful information.
+
+        Note:
+            In general, no guarantee is given that all elements are present, so
+            use .get() functions with defaults if processing these elements of
+            run_param_dict.
+        """
+        if self.run_param_dict is None:
+            self.run_param_dict = {}
+        # Main run info
+        self.run_param_dict['datetime'] = datetime.datetime.utcnow()
+        self.run_param_dict['nproc'] = mwxrun.n_procs
+        # TODO: Implement when needed
+        # self.run_param_dict['decomp'] = warp.warpoptions.options.decomp
+        if self.run_file is not None:
+            with open(self.run_file, 'r') as rfile:
+                self.run_param_dict['run_file'] = rfile.read()
+        # Other info
+        self._save_version_info()
+        self._save_comp_info()
+
+    def _save_version_info(self):
+        """Save info about file versions."""
+        # self.run_param_dict['metools_version'] = metools.__version__
```

Comment:
```suggestion
    def _save_version_info(self):
        """Save info about file versions."""
        raise NotImplementedError
        # self.run_param_dict['metools_version'] = metools.__version__
```
I think moving `NotImplementedErrors` to before commented out code is cleaner, Peter has had me do that before.

Reply:
Also a comment for the other PR.